graylog2-server/src/main/java/org/graylog/security/entities/EntityDependencyResolver.java

```diff
@@ -42,6 +46,11 @@
     private final GRNRegistry grnRegistry;
     private final GRNDescriptorService descriptorService;
     private final DBGrantService grantService;
+    // Some dependencies can be ignored.
+    // E.g. To view a stream with a custom output, a user does not need output permissions
+    private static final Map<GRNType, List<ModelType>> IGNORED_DEPENDENCIES = ImmutableMap.<GRNType, List<ModelType>>builder()
```

Comment:
How about using a `Set<ModelType>` here? This should make the contains check more efficient and I think we don't need the ordering properties of a list here.

Reply:
Yeah, makes sense :+1: